lib/Log.js

```diff
@@ -209,9 +211,11 @@ class Log {
                 return this._helper.logStream(
                     this._deviceIds, 
                     this._messageHandler.bind(this),
-                    this._stateChangeHandler.bind(this)).
+                    this._stateChangeHandler.bind(this),
+                    this._errorHandler.bind(this)).
                     then((logStreamId) => {
                         this._logStreamId = logStreamId;
+                        this._reconnecting = false;
```

Comment:
We only toggle the flag on success. What if we fail to reconnect? Are we just exiting in this case?

Reply:
Yes, if we fail to reconnect we are just exit through _reconnect method error handler.